plugin/src/test/groovy/com/novoda/test/TestAndroidKotlinProject.groovy

```diff
@@ -0,0 +1,78 @@
+package com.novoda.test
+
+class TestAndroidKotlinProject extends TestProject<TestAndroidKotlinProject> {
+    private static final Closure<String> TEMPLATE = { TestAndroidKotlinProject project ->
+        """
+buildscript {
+    repositories { 
+        google()
+        jcenter()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.0.1'  
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.20'
+    }
+}
+plugins {
+    ${formatPlugins(project)}
+}
+repositories { 
+    google()
+    jcenter()
+}
+apply plugin: 'com.android.library'  
+apply plugin: 'kotlin-android'
+
+android {
+    compileSdkVersion 27
+    buildToolsVersion '27.0.0'
+
+    defaultConfig {
+        minSdkVersion 16
+        targetSdkVersion 27
+        versionCode 1
+        versionName '1.0'
+    }
+    sourceSets {
+        ${formatSourceSets(project)}
+    }
+    ${project.additionalAndroidConfig}
+}
+${formatExtension(project)}
+"""
+    }
+
+    private String additionalAndroidConfig = ''
+
+    TestAndroidKotlinProject() {
+        super(TEMPLATE)
+        File localProperties = Fixtures.LOCAL_PROPERTIES
+        if (localProperties.exists()) {
+            withFile(localProperties, 'local.properties')
+        }
+    }
+
+    private static String formatSourceSets(TestProject project) {
+        project.sourceSets
+                .entrySet()
+                .collect { Map.Entry<String, List<String>> entry ->
+            """$entry.key {
+            manifest.srcFile '${Fixtures.ANDROID_MANIFEST}'
+            kotlin {
```

Comment:
This has been changed

Reply:
well, if you add your sources to `java` folder, you don't need this in test setup.